app/src/test/java/org/mozilla/fenix/home/sessioncontrol/viewholders/topsites/TopSitesPagerAdapterTest.kt

```diff
@@ -4,44 +4,180 @@
 
 package org.mozilla.fenix.home.sessioncontrol.viewholders.topsites
 
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
 import mozilla.components.feature.top.sites.TopSite
+import mozilla.components.feature.top.sites.TopSite.Type.DEFAULT
 import org.junit.Assert.assertEquals
+import org.junit.Before
 import org.junit.Test
-import org.mozilla.fenix.home.sessioncontrol.AdapterItem
+import org.mozilla.fenix.home.sessioncontrol.AdapterItem.TopSitePagerPayload
 
 class TopSitesPagerAdapterTest {
```

Comment:
These are super valuable tests and it would be nice to have one that looks at the other adapter as a blackbox to assert the changes we expect. That would really help us avoid regressions but also nail down or expectations of how the results of the adapter should behave.

Something like:

```
topSitesPagerAdapter.submitList(listOfTopSites)
val result = topSitesPagerAdapter.getPagedTopSites() // List<Int, List<TopSite>>

assert(firstTopSite, result[0].first())
assert(secondTopSite, result[1].first())
// etc..
```

The hope is that, if we change the internals, like that nested RecyclerView structure, we can rely on a test like this to verify we haven't goofed. Would be great to know what you think about it. 🙂

Reply:
This is a very good idea, I'll try and implement it! 👍